composeApp/src/iosMain/kotlin/org/ooni/probe/background/OperationsManager.kt

```diff
@@ -1,26 +1,57 @@
 package org.ooni.probe.background
 
 import co.touchlab.kermit.Logger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.async
 import org.ooni.probe.data.models.InstalledTestDescriptorModel
 import org.ooni.probe.data.models.RunSpecification
 import org.ooni.probe.di.Dependencies
 import platform.BackgroundTasks.BGProcessingTask
+import platform.Foundation.NSDate
 import platform.Foundation.NSOperationQueue
+import platform.Foundation.timeIntervalSinceDate
 import platform.UIKit.UIApplication
 
 class OperationsManager(private val dependencies: Dependencies, private val backgroundRunner: BackgroundRunner) {
+    // https://developer.apple.com/documentation/foundation/nsdate/1410206-timeintervalsince
     fun startSingleRun(spec: RunSpecification) {
+        val startDateTime = NSDate()
+        val differed = CoroutineScope(Dispatchers.Default).async {
+            try {
+                dependencies.runBackgroundTask(spec).collect()
+                Logger.i { "Job finished successfully in ${NSDate().timeIntervalSinceDate(startDateTime)} ms" }
+            } catch (e: Exception) {
+                Logger.e(e) { "Job failed with exception in ${NSDate().timeIntervalSinceDate(startDateTime)} ms" }
+            }
+        }
         backgroundRunner(background = {
-            runBlocking { dependencies.runBackgroundTask(spec).collect() }
+            differed.start()
+        }, cancel = {
+            differed.cancel()
+            Logger.i { "Job cancelled after ${NSDate().timeIntervalSinceDate(startDateTime)} ms" }
```

Comment:
`Logger.i` will only log as a breadcrumb. This only gets sent if there's an error. Or will we always call line 26 inside the `catch` when it's aborted?

Reply:
Nope only on error.